### PR TITLE
Fix CMake FindPDAL for PDAL version 2.6+

### PR DESCRIPTION
**Pull request by nilason**
## Description

The newly released PDAL 2.6 removed the 'pdal_util' library as a separate library, it is now part of the 'pdalcpp' library (see https://github.com/PDAL/PDAL/releases/tag/2.6.0). The CMake FindPDAL thus fails to locate this now non-existing library. This PR addresses this issue and should be backported to 3.28 and 3.32.

Note, I added a style formatting commit for the FindPDAL.cmake file to this PR  with the intent to modernise the code and improve readability. If you find that particular change is not appropriate, I'll remove that part.

### CODE REVIEW

**Comment by m-kuhn** [comment]
Don't newer PDAL version already ship a pdalconfig.cmake file which we could rely on and only use the old code as fallback?
Cmp. what has been done for Proj etc https://github.com/qgis/QGIS/commit/789e92b05436f3d1f6d3ca8950a74265892090d3#diff-2fa2a6c4b1439910cf5be5f527f911fd4c752870c0b999306dbdcd61df63211bL21

**Comment by nilason** [comment]
> Don't newer PDAL version already ship a pdalconfig.cmake file which we could rely on and only use the old code as fallback? Cmp. what has been done for Proj etc [789e92b#diff-2fa2a6c4b1439910cf5be5f527f911fd4c752870c0b999306dbdcd61df63211bL21](https://github.com/qgis/QGIS/commit/789e92b05436f3d1f6d3ca8950a74265892090d3#diff-2fa2a6c4b1439910cf5be5f527f911fd4c752870c0b999306dbdcd61df63211bL21)

Actually, PDAL did almost always come with cmake config-files. Thanks for the Proj solution link. I have now updated this PR accordingly. Tested with PDAL 2.5.6 and 2.6.0.

I wonder if  the:
```cmake
if (WITH_PDAL AND MSVC)
  add_definitions(-DWIN32_LEAN_AND_MEAN)
endif()
```
parts cannot be moved to FindPDAL.cmake? It is now repeated in three files.

File: src/app/CMakeLists.txt [hunk 1]
```diff
@@ -482,11 +482,8 @@ if(HAVE_OPENCL)
     include_directories(SYSTEM ${OpenCL_INCLUDE_DIRS})
 endif()
 
-if (WITH_PDAL)
-  include_directories(SYSTEM ${PDAL_INCLUDE_DIR})
-  if(MSVC)
-    add_definitions(-DWIN32_LEAN_AND_MEAN)
-  endif()
+if (WITH_PDAL AND MSVC)
+  add_definitions(-DWIN32_LEAN_AND_MEAN)
```
Comment (m-kuhn) [inline src/app/CMakeLists.txt:486]: I guess this should go to FindPDAL as well with 
[`target_compile_definitions`](https://cmake.org/cmake/help/v3.2/command/target_compile_definitions.html)

Comment (nilason) [reply src/app/CMakeLists.txt:486]: That would probably not work as the target *is* imported:
> The named <target> must have been created by a command such as [add_executable()](https://cmake.org/cmake/help/v3.2/command/add_executable.html#command:add_executable) or [add_library()](https://cmake.org/cmake/help/v3.2/command/add_library.html#command:add_library) and must not be an [Imported Target](https://cmake.org/cmake/help/v3.2/manual/cmake-buildsystem.7.html#imported-targets).

Comment (nilason) [reply src/app/CMakeLists.txt:486]: It might work for the fallback code. Wouldn't that part be included with the config-files for Windows?

Comment (m-kuhn) [reply src/app/CMakeLists.txt:486]: Yes, that should be part of the fallback code or defined by the config files in the Windows case

Comment (nilason) [reply src/app/CMakeLists.txt:486]: > ...  or defined by the config files in the Windows case

Well, is it?

And is `WIN32_LEAN_AND_MEAN` needed at all nowadays (just looked at this 14-year-old post https://devblogs.microsoft.com/oldnewthing/20091130-00/?p=15863)?
Maybe @jef-n can help us here?

Comment (jef-n) [reply src/app/CMakeLists.txt:486]: it is: https://cdash.orfeo-toolbox.org/build/18183

File: cmake/FindPDAL.cmake [hunk 2]
```diff
@@ -7,8 +7,7 @@
 # CMake module to search for PDAL library
 #
 # If it's found it sets PDAL_FOUND to TRUE
-# and following variables are set:
-#    PDAL_INCLUDE_DIR
+# and adds the following target(s):
 #    PDAL_LIBRARIES
```
Comment (m-kuhn) [inline cmake/FindPDAL.cmake:11]:
```suggestion
#    - pdalcpp
#    - pdal_util
```

File: src/app/CMakeLists.txt (hunk 1 above)
Comment (nilason) [reply src/app/CMakeLists.txt:486]: > That would probably not work as the target _is_ imported:
> 
> > The named `<target>` must have been created by a command such as [add_executable()](https://cmake.org/cmake/help/v3.2/command/add_executable.html#command:add_executable) or [add_library()](https://cmake.org/cmake/help/v3.2/command/add_library.html#command:add_library) and must not be an [Imported Target](https://cmake.org/cmake/help/v3.2/manual/cmake-buildsystem.7.html#imported-targets).

That quote was from a very old cmake version docs. If I'm not mistaken, minimal CMake version required for QGIS is 3.12. In that case `target_compile_definitions` seems to work.

File: cmake/FindPDAL.cmake (hunk 2 above)
Comment (nilason) [reply cmake/FindPDAL.cmake:11]: The PDAL config-files defines `PDAL_LIBRARIES`, which may be either a list `pdalcpp;pdal_util` (<2.6) or only `pdalcpp` (2.6+).  So PDAL_LIBRARIES should stay, but I may add a clarifying comment.

Comment (m-kuhn) [reply cmake/FindPDAL.cmake:11]: Good point, let's keep as it is then 👍

File: src/app/CMakeLists.txt (hunk 1 above)
Comment (nilason) [reply src/app/CMakeLists.txt]: I added a commit with all windows definitions of WIN32_LEAN_AND_MEAN moved to FindPDAL (using target_compile_definitions).

**Review by m-kuhn — APPROVED** [review]

**Review by lbartoletti — APPROVED** [review]

**Comment by agiudiceandrea** [comment]
It looks like this PR should have been backported to the release-3_28 brach: please see https://github.com/qgis/QGIS/issues/55431.

**Comment by imincik** [comment]
Is it possible to backport this change to `release-3_28` so we can build QGIS LTR with PDAL 2.6 ? Thanks

**Pull request merged**